darts/models/theta.py

```diff
@@ -115,14 +123,228 @@ def predict(self, n: int) -> 'TimeSeries':
         return 'Theta({})'.format(self.theta)
+
+
+class FourTheta(UnivariateForecastingModel):
+    def __init__(self,
+                 theta: int = 2,
+                 seasonality_period: Optional[int] = None,
+                 model_mode: Model = Model.ADDITIVE,
+                 season_mode: Season = Season.MULTIPLICATIVE,
+                 trend_mode: Trend = Trend.LINEAR):
+        """
+        An implementation of the 4Theta method with configurable `theta` parameter.
+
+        See M4 competition `solution <https://github.com/Mcompetitions/M4-methods/blob/master/4Theta%20method.R>`_.
+
+        The training time series is de-seasonalized according to `seasonality_period`,
+        or an inferred seasonality period.
+
+        This model is similar to Theta, with `theta` = 2-theta, `model_mode` = additive and `trend_mode` = linear.
+
+        Parameters
+        ----------
+        theta
+            Value of the theta parameter. Defaults to 2.
+            If theta = 1, then the theta method restricts to a simple exponential smoothing (SES)
+        seasonality_period
+            User-defined seasonality period. If not set, will be tentatively inferred from the training series upon
+            calling `fit()`.
+        model_mode
+            Type of model combining the Theta lines. Either "additive" or "multiplicative".
+        season_mode
+            Type of seasonality. Either "additive" or "multiplicative".
+        trend_mode
+            Type of trend to fit. Either "linear" or "exponential".
+        """
+
+        super().__init__()
+
+        self.model = None
+        self.drift = None
+        self.coef = 1
+        self.mean = 1
+        self.length = 0
+        self.theta = theta
+        self.is_seasonal = False
+        self.seasonality = None
+        self.season_period = seasonality_period
+        self.model_mode = model_mode
+        self.season_mode = season_mode
+        self.trend_mode = trend_mode
+        self.wses = 0 if self.theta == 0 else (1 / theta)
+        self.wdrift = 1 - self.wses
+        self.fitted_values = None
+
+        raise_if_not(model_mode in Model,
+                     "Unknown value for model_mode: {}.".format(model_mode), logger)
+        raise_if_not(trend_mode in Trend,
+                     "Unknown value for trend_mode: {}.".format(trend_mode), logger)
+        raise_if_not(season_mode in Season,
+                     "Unknown value for season_mode: {}.".format(season_mode), logger)
+
+        # Remark on the values of the theta parameter:
+        # - if theta = 1, then the theta method restricts to a simple exponential smoothing (SES)
+        # - if theta = 0, then the theta method restricts to a simple `trend_mode` regression.
+
+    def fit(self, ts, component_index: Optional[int] = None):
+        super().fit(ts, component_index)
+
+        self.length = len(ts)
+        # normalization of data
+        self.mean = ts.mean().mean()
+        if self.mean == 0:
```

Comment:
That seems weird, do they do the same thing in the original implementation?

Reply:
The TimeSeries in M4 are all stricly positive, so no.
I will raise an error in this case, but the model can run well without normalization in this case.

Reply:
Additionally, I can let the choice to use the normalization to the user with a parameter.